subsys/bluetooth/host/crypto.c

```diff
@@ -91,9 +92,13 @@ int prng_init(void)
 	}
 
 	/* re-seed is needed after init */
-	return prng_reseed(&prng);
+	ret = prng_reseed(&prng);
+#endif /*CONFIG_BT_HOST_CRYPTO_RANDOM*/
+
+	return ret;
```

Comment:
The change here is to simply return 0 if `CONFIG_BT_HOST_CRYPTO_RANDOM=n`. Would it make more sense to just guard the entire function by `CONFIG_BT_HOST_CRYPTO_RANDOM`, so that it isn't callable?

Reply:
We need to check whether the controller supports the BT RANDOM command.

Reply:
Why? 

if `!IS_ENABLED(CONFIG_BT_HOST_CRYPTO_RANDOM` then it doesn't really matter if the controller supports it or not, right? Because it will just return 0 in any case, and not perform any actions.

Reply:
Add `BT_CMD_TEST(bt_dev.supported_commands, 27, 7)` check. I don't think it's unnecessary. After all, in the later operation, the bt random command will be issued directly https://github.com/zephyrproject-rtos/zephyr/blob/69829d2a5698c7621c1006bbec2d953017b56e29/subsys/bluetooth/host/crypto.c#L132

Reply:
Without this check, it is possible to issue a command that is not supported.

Reply:
Not sure I've expressed myself precisely. 

If `CONFIG_BT_HOST_CRYPTO_RANDOM=n` then the function does nothing, except return "success", unless the controller does not support it. 

Why do we need to check controller support if we are not going to use the controller?